File: 알고리즘/2/Longest Palindrome.py

class Solution:
    def longestPalindrome(self, s: str) -> int:
        dict1 = dict()

        for ele in s:
            dict1[ele] = dict1.get(ele, 0) + 1

        cnt = 0
        flag = False
        for key, value in dict1.items():
            cnt += (value//2*2)

            if value % 2 == 1:
                flag = True

        if flag:
            cnt += 1

        return cnt


s = Solution()
print(s.longestPalindrome('abccccdd'))
print(s.longestPalindrome('a'))
print(s.longestPalindrome('ccc'))

File: 알고리즘/2/Longest Palindrome.cpp
#include <bits/stdc++.h>

using namespace std;
class Solution
{
public:
    int longestPalindrome(string s)
    {
        unordered_map<char, int> map;

        for (auto ele : s)
        {
            map[ele] += 1;
        }

        int cnt = 0;

        bool flag = false;
        for (auto iter = map.begin(); iter!=map.end(); iter++){
            cnt+=(iter->second/2)*2;
            if (iter->second %2 ==1) flag = true;
        }

        if(flag){
            cnt+=1;
        }
        return cnt;
    }
};